### PR TITLE
refactor(digital-twin-client): Make the DigitalTwin client mockable.

### PR DESCRIPTION
**Pull request by vinagesh**
## Checklist
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-iot-sdk-csharp/blob/master/.github/CONTRIBUTING.md).
- [ ] I added or modified the existing tests to cover the change (we do not allow our test coverage to go down).
- [ ] This pull-request is submitted against the `master` branch.

### CODE REVIEW

**Review by drwill-ms — APPROVED** [review]

File: iothub/service/src/DigitalTwin/DigitalTwinClient.cs [hunk 1]
```diff
@@ -21,6 +21,13 @@ public class DigitalTwinClient : IDisposable
         private readonly IotHubGatewayServiceAPIs _client;
         private readonly DigitalTwin _protocolLayer;
 
+        /// <summary>
+        /// Creates an instance of <see cref="DigitalTwinClient"/>
```
[the remark this answers is not in the document]
Comment (vinagesh) [reply iothub/service/src/DigitalTwin/DigitalTwinClient.cs:25]: yes there should be. I Just added it while you were adding this comment :)

**Pull request merged**